tensorflow_serving/batching/batching_session.cc

```diff
@@ -340,7 +354,12 @@ Status BatchingSession::MergeInputTensors(
 
   // For each input tensor name, a vector of tensors from the individual tasks.
   std::map<string, std::vector<Tensor>> tensors_to_merge;
-
+  std::map<string, std::vector<int>> max_dim_sizes;
```

Comment:
Since you only populate/use this if padding is requested, can you make it an optional<map>, so it defaults to "no value set" rather than an empty map.

See tensorflow_serving/util/optional.h.